lib/jobs/uefi-job.js

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = uefiJobFactory;
+di.annotate(uefiJobFactory, new di.Provide('Job.Uefi'));
+di.annotate(uefiJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'Logger',
+        'Assert',
+        'Util'
+    )
+);
+function uefiJobFactory(BaseJob, Logger, assert, util) {
+    var logger = Logger.initialize(uefiJobFactory);
+
+    /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function UefiJob(options, context, taskId) {
+        UefiJob.super_.call(this, logger, options, context, taskId);
+        assert.string(options.uefitool);
+        assert.string(options.repo);
+        this.nodeId = this.context.target;
+        this.profile = this.options.profile;
+    }
+    util.inherits(UefiJob, BaseJob);
+
+    UefiJob.prototype._run = function () {
+        var self = this;
+        self._subscribeRequestProfile(function () {
+            return self.profile;
```

Comment:
Since the `profile` is in the base-task's `requiredOptions` list and it is a must-have options for `subscribeRequestProfile`, so I suggest adding an assertion logic for the existing of `profile`